py/farm_ng/core/events_file_reader.py

```diff
@@ -45,10 +55,8 @@ def __init__(self, file_name: Union[str, Path]) -> None:
         self._file_stream: Optional[IO] = None
         self._file_length: int = 0
 
-        # store the bytes offset in a dictionary where:
-        # - the key is the string representation of the Uri of the message
-        # - the value is a list with tuples of Event and its offsets to messages with that Uri
-        self._event_index: DefaultDict[str, List[Tuple[Event, int]]] = defaultdict(list)
+        # store index of events in a list to seek faster
+        self._event_index: List[EventLogPosition] = []
```

Comment:
Clearer that it is all events, not the index of a single event
```suggestion
        self._events_index: List[EventLogPosition] = []
```